### PR TITLE
Fixed returns in the sacl functions to differentiate between errors

### PR DESCRIPTION
**Pull request by jotacarma90**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/6419|

## Description

This PR fixes how the functions `check_object_sacl` and `is_valid_sacl` work with the returns values, to make the functionality of error handling more accurate.

### is_valid_sacl
Return:
- 0 if sacl is valid
- 1 if sacl is invalid or NULL
- 2 if cannot initialize **everyone_sid**

### check_object_sacl
Return:
- 2 if fail some of preparation
- the same return obtained in **is_valid_sacl**

## Tests

Fixed unit test related to these functions.

### CODE REVIEW

**Comment by jotacarma90** [comment]
Integration test in Jenkins passed:
[agent_ubuntu_html_report_test_integration_B1399_20210305191820.zip](https://github.com/wazuh/wazuh/files/6107935/agent_ubuntu_html_report_test_integration_B1399_20210305191820.zip)
[agent_solaris_html_report_test_integration_B1400_20210305191820 (1).zip](https://github.com/wazuh/wazuh/files/6107937/agent_solaris_html_report_test_integration_B1400_20210305191820.1.zip)
[agent_macos_html_report_test_integration_B1400_20210305191820.zip](https://github.com/wazuh/wazuh/files/6107938/agent_macos_html_report_test_integration_B1400_20210305191820.zip)
[agent_windows_html_report_test_integration_B1399_20210305191820.zip](https://github.com/wazuh/wazuh/files/6107939/agent_windows_html_report_test_integration_B1399_20210305191820.zip)
[agent_centos_html_report_test_integration_B1399_20210305191820.zip](https://github.com/wazuh/wazuh/files/6107940/agent_centos_html_report_test_integration_B1399_20210305191820.zip)

There are some failures in the Ubuntu build, they should not since the changes only affect Windows. Reviewing.

File: src/syscheckd/whodata/win_whodata.c [hunk 1]
```diff
@@ -150,27 +150,28 @@ int set_winsacl(const char *dir, int position) {
     ZeroMemory(&old_sacl_info, sizeof(ACL_SIZE_INFORMATION));
 
     // Check if the sacl has what the whodata scanner needs
-    switch(is_valid_sacl(old_sacl, (syscheck.wdata.dirs_status[position].object_type == WD_STATUS_FILE_TYPE) ? 1 : 0)) {
-        case 0:
-            mdebug1(FIM_SACL_CHECK_CONFIGURE, dir);
-            syscheck.wdata.dirs_status[position].status |= WD_IGNORE_REST;
+    switch (is_valid_sacl(old_sacl, (syscheck.wdata.dirs_status[position].object_type == WD_STATUS_FILE_TYPE) ? 1 : 0)) {
```
Comment (Molter73) [inline src/syscheckd/whodata/win_whodata.c:153]: I know this was not changed in this PR but is really hard to understand, could we refactor it to something like this? The definition of `is_file` can be moved up in the function.
```suggestion
    int is_file = syscheck.wdata.dirs_status[position].object_type == WD_STATUS_FILE_TYPE) ? 1 : 0;
    
    switch (is_valid_sacl(old_sacl, is_file) {
```

**Review by Molter73 — CHANGES_REQUESTED** [review]
It would be greatly appreciated if we could add doxygen documentation to the functions being modified. They return integers and is a little hard to follow exactly what each number is supposed to mean. Other than that and a minor refactor, the PR is looking good.

**Review by Molter73 — DISMISSED** [review]
The PR is looking good, just a minor style change and a fix in a doxygen description. The PR might need a rebase to pass Jenkins tests but other than that, it should be good to go.

**Comment by jotacarma90** [comment]
Integration test on Jenkins:
[agent_windows_html_report_test_integration_B1562_20210408160638.zip](https://github.com/wazuh/wazuh/files/6284376/agent_windows_html_report_test_integration_B1562_20210408160638.zip)
[agent_ubuntu_html_report_test_integration_B1562_20210408160638.zip](https://github.com/wazuh/wazuh/files/6284377/agent_ubuntu_html_report_test_integration_B1562_20210408160638.zip)
[agent_centos_html_report_test_integration_B1562_20210408160638.zip](https://github.com/wazuh/wazuh/files/6284378/agent_centos_html_report_test_integration_B1562_20210408160638.zip)

**Review by JoseAntonioMHerrera — APPROVED** [review]
